src/base/bittorrent/torrenthandleimpl.cpp

```diff
@@ -1368,6 +1368,37 @@ void TorrentHandleImpl::renameFile(const int index, const QString &name)
     m_nativeHandle.rename_file(lt::file_index_t {index}, Utils::Fs::toNativePath(name).toStdString());
 }
 
+void TorrentHandleImpl::renameFolder(const QString &oldPath, const QString &newPath)
```

Comment:
`TorrentContentTreeView::renameSelectedFile` can also make use of this function if you're feeling up for it.

Reply:
This has been completed in: https://github.com/qbittorrent/qBittorrent/pull/13081/commits/24f505c59eaa5f77d3dd2ca7a21b0124f8332068

However, I had one question. In the above commit, I felt the urge to do the following:
```c++
void TorrentHandleImpl::renameFolder(const QString &oldPath, const QString &newPath)
{
    info().renameFolder(oldPath, newPath);
}
```

I opted to keep the code duplicated/separated. My thoughts are they were separated for a reason that has not been shared with me yet. Should I go back in and remove the duplicated logic?

Reply:
>`info().renameFolder(oldPath, newPath);`

IIRC, it won't work.

Reply:
Okay then I will leave it as is, unless instructed otherwise.